contrib/devtools/security-check.py

```diff
@@ -176,6 +176,18 @@ def check_NX(binary) -> bool:
     '''
     Check for no stack execution
     '''
+
+    # The PPC64 ABI uses a non-executable stack by default, and does not
+    # need a GNU_STACK program header.
+    # This can be dropped once LIEF is improved upstream:
+    # https://github.com/lief-project/LIEF/issues/717
+    arch = binary.abstract.header.architecture
+    binary.concrete
```

Comment:
This line looks really strange.

Reply:
Yes, this is a little awkward, and was what I was alluding to in the PR description. Ideally, I'd use `if binary.header.machine_type == lief.ELF.ARCH.PPC64:` here, but because this function is used across all binaries, not just ELF, that wont work, as, for example, the PE binary header doesn't have a `machine_type`. Instead, we use `binary.abstract.header.architecture`, however whenever you use `.abstract`, you have to use `.concrete` to case the binary back from the abstract type. See here: https://lief-project.github.io//doc/latest/api/python/abstract.html#binary.

Reply:
Hm, might want to add a comment to that regard in-line.

Reply:
> Or always use binary.abstract or binary.concrete explicitly?

I'm not even sure that'd be possible, as you've got access to different functionality in either. In any case, a wholesale change like that isn't going to be part of this PR.

Reply:
It's a temporary work-around, changing the whole code for it is probably not worth it (unless it's needed for other things too).